diff --git a/src/utarray.h b/src/utarray.h
index cc20614..d018c44 100644
--- a/src/utarray.h
+++ b/src/utarray.h
@@ -97,7 +97,7 @@ typedef struct {
 #define utarray_reserve(a,by) do {                                            \
   if (((a)->i+(by)) > (a)->n) {                                               \
     char *utarray_tmp;                                                        \
-    while (((a)->i+(by)) > (a)->n) { (a)->n = ((a)->n ? (2*(a)->n) : 8); }    \
+    while (((a)->i+(by)) > (a)->n) { (a)->n = ((a)->n ? (2*(a)->n) : 4); }    \
     utarray_tmp=(char*)realloc((a)->d, (a)->n*(a)->icd.sz);                   \
     if (utarray_tmp == NULL) {                                                \
       utarray_oom();                                                          \
